unmarshal_from_json_map.go

```diff
@@ -80,7 +80,7 @@ func (m *mapDecoder) populateStruct(path []string, data map[string]interface{},
 					field := refInfo.field(structValue)
 					assignValue(field, value)
 				}
-				if result != nil {
+				if result != nil && !m.options.excludeKnownFieldsFromMap {
```

Comment:
Same comment as the if-else block above

Reply:
fixed. moved the condition inside the if block